src/blockdata/transaction.rs

```diff
@@ -343,6 +347,15 @@ impl Transaction {
         let sighash_type: u32 = sighash_type.into();
         assert!(input_index < self.input.len());  // Panic on OOB
 
+        if self.is_invalid_use_of_sighash_single(sighash_type, input_index) {
+            // We cannot correctly handle the SIGHASH_SINGLE bug here because usage of this function
+            // will result in the data written to the writer being hashed, however the correct
+            // handling of the SIGHASH_SINGLE bug is to return the 'one array' - either implement
+            // this behaviour manually or use `signature_hash()`.
+            writer.write(b"[not a transaction] SIGHASH_SINGLE bug")?;
+            return Ok(())
```

Comment:
Either this should be returning an Err(encode::Error::SigHashSingle) or the error enum variant should be removed.

Reply:
Thanks man, sloppy work by me leaving that in - no excuse.

Reply:
No worries, happens to me a lot :)